db/seeds.d/20-discovery-roles.rb

```diff
@@ -0,0 +1,31 @@
+class DiscoveryPermissions
+  READER = [
+    :view_discovered_hosts,
+    :view_discovery_rules,
+    :view_locations,
+    :view_organizations,
+  ]
+
+  MANAGER = READER + [
+    :assign_organizations,
+    :assign_locations,
+    # discovered_hosts
+    :submit_discovered_hosts,
+    :auto_provision_discovered_hosts,
+    :provision_discovered_hosts,
+    :edit_discovered_hosts,
+    :destroy_discovered_hosts,
+    # discovered_rules
+    :create_discovery_rules,
+    :edit_discovery_rules,
+    :execute_discovery_rules,
+    :destroy_discovery_rules,
+  ]
+end
+
+unless Rails.env.test?
+  Role.without_auditing do
+    create_role('Discovery Reader', DiscoveryPermissions::READER, 0)
+    create_role('Discovery Manager', DiscoveryPermissions::MANAGER, 0)
+  end
+end
```

Comment:
@domcleal does this look reasonable as the standard for other plugins?

This deprecates role DSL and solves two issues:

1) Chicken-n-egg problem for permissions from core you want to assign to plugin roles
2) Possibility to test permissions in functional tests (we load this file in a test helper)
3) Ability to load this the same way from Foreman Core to add Role validation feature later on